### DISCUSSION TITLE
How to save features extracted by the backbone

### DISCUSSION THREAD

**Discussion by CLeaf1023** [open-edge-platform/anomalib#428] [Q&A] [answered]
I tried to save feature in ‘’anomalib/models/components/feature_extractors/feature_extractor.py‘’
![图片](https://user-images.githubusercontent.com/31176427/177107447-1f1be39e-8d8f-457b-85f2-2849674b1f1e.png)
but it displays that the feature shape is
![图片](https://user-images.githubusercontent.com/31176427/177107761-36e63818-eeab-4ba8-8af8-42acaad0ff7c.png)

it seems like in this part, feature hasn't been generated.
if i want to save the features extracted by the backbone,which python file should i edit?

**Comment by ashwinvaidya17** [comment, marked as answer]
You should call the print statement after `_ = self.backbone(input_tensor)` line (95). It is because hooks are added to the backbone and only when the forward function of the backbone is called, the features dictionary is populated.

**Comment by samet-akcay** [comment]
Converting to Q&A in Discussions, feel free to continue from there.
